support backfill id filtering for backfills

jamiedemaria opened the pull request:
## Summary & Motivation
Adds a `backfill_ids` filter to `BulkActionsFilter` so that the `backfill id` filter option on the runs feed can operate on backfills too.

I originally considered just adding a special case when fetching the runs feed where if the backfill id filter was passed we directly fetched the backfill. however, that required manually checking that all of the other filters matched the backfill, and that the cursor was respected, which felt prone to bugs. You can see the draft here if interested https://github.com/dagster-io/dagster/pull/24793

corresponding internal pr https://github.com/dagster-io/internal/pull/11818

## How I Tested These Changes

## Changelog

NOCHANGELOG

jamiedemaria commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24799?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24799** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24799?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#24745** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24745?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#24755** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24755?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#24754** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24754?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jamiedemaria and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

jamiedemaria commented on python_modules/dagster/dagster/_core/execution/backfill.py:
```diff
@@ -62,13 +62,15 @@ class BulkActionsFilter:
     created_after: Optional[datetime] = None
     tags: Optional[Mapping[str, Union[str, Sequence[str]]]] = None
     job_name: Optional[str] = None
+    backfill_ids: Optional[Sequence[str]] = None
```
BulkActionsFilter isn't part of the public api, but i could still mark the new param as experimental. of the ones i've added recently, this one seems the most likely to be removed/changed

OwenKephart replied:
it's not public, so I think we're fine here

prha reviewed (dismissed):
can we split apart the core changes and the graphql changes?

do we not need gating for the internal storage tests which presumably run the run storage test suite?

jamiedemaria commented:
> do we not need gating for the internal storage tests which presumably run the run storage test suite?

i implemented the internal support for the filter here https://github.com/dagster-io/internal/pull/11818, so is the idea that the gating would help in the case that we reverted the internal changes?

prha commented:
well, wondering if we needed to support the behavior of the filter in the ursula storage.

OwenKephart commented on python_modules/dagster/dagster_tests/storage_tests/utils/run_storage.py:
```diff
@@ -1730,6 +1730,38 @@ def test_backfill_simple_job_name_filtering(self, storage: RunStorage):
+            tags={},
+            backfill_timestamp=time.time(),
+        )
+        storage.add_backfill(backfill)
```
nitpick, but could we also add another backfill with a different id just to make sure this filter is actually applied?

OwenKephart reviewed (approved):
note comment

OwenKephart commented:
> well, wondering if we needed to support the behavior of the filter in the ursula storage.

whoops approved before seeing that, I assume this will fail in the internal test suite for that one?
